client/src/components/Main/Main.jsx

```diff
@@ -1,21 +1,61 @@
-import React from 'react';
-import { useSelector } from 'react-redux';
-import { Switch, Route, Redirect } from 'react-router-dom';
+/* eslint-disable react/prop-types */
+/* eslint-disable react/jsx-props-no-spreading */
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Switch, Route, withRouter } from 'react-router-dom';
+import axios from 'axios';
 import { Box } from '@material-ui/core';
 import Header from '../Header/Header';
-import AdminLogin from '../../pages/AdminLogin';
-import Dashboard from '../../pages/Dashboard';
-import CreateSurvey from '../../pages/CreateSurvey';
-import EditSurvey from '../../pages/EditSurvey/index';
-import SurveyDetail from '../../pages/SurveyDetail';
-import SurveyBuilderFromTemplate from '../../pages/SurveyBuilderFromTemplate';
-import TakeSurvey from '../../pages/TakeSurvey';
+
 import PageNotFound from '../../pages/PageNotFound';
+import LoadingPageOrRedirect from './LoadingPageOrRedirect'
+import { routes, protectedRoutes } from './routes'
+
+import addTokenToEveryRequest from '../../utils/addAuthorizationHeaderToEveryRequest';
+import deleteTokenOn401StatusCodes from '../../utils/deleteTokenOn401StatusCodes';
+import checkTokenIsAuth from '../../utils/checkTokenIsAuth';
+import { CHECKING_IF_AUTHED } from '../../store/actions/mainActions';
+
+const Main = ({ history }) => {
+  const dispatch = useDispatch();
 
-const Main = () => {
   const { data } = useSelector((state) => state.adminLoginReducer);
   const { auth } = data;
   const { snackbar } = useSelector((state) => state.snackbarReducer);
+  const { checkingIfAuthed } = useSelector((state) => state.mainReducer);
```

Comment:
given this isn't a function I don't think its name should be a verb. Can we just call it `userAuthorised` or similar?